include/swift/AST/TensorFlow.h

```diff
@@ -55,10 +55,16 @@ namespace tf {
   /// VariantHandle.
   bool isTensorFlowValue(Type ty);
 
-  /// Returns true if the specified type is a TensorFlow value or an tuple or
+  /// Return true if the specified type is a TensorFlow value or an tuple or
   /// struct of such.
   bool isTensorFlowValueOrAggregate(Type ty);
 
+  /// Flatten an aggregate of TensorFlow value types and push them to the result
+  /// array. If the given type itself is a TensorFlow value type, it will be
+  /// pushed to the array. If any terminal type is not a TensorFlow value type,
+  /// return false and the result will be empty.
+  bool flattenTensorFlowValueAggregate(Type ty, SmallVectorImpl<Type> &result);
```

Comment:
Can this use the function introduced in Nanjiang's patch?

Reply:
@njjiang's function is collecting dtypes (extracting the `Scalar` from `TensorHandle`'s), while this function is collecting TensorHandle's. I could generalize this logic somehow, but it might make the interface super complex.

Reply:
ok!

Reply:
ok -> makes sense, don't make it super complex :-)